slack_sdk/models/blocks/basic_components.py

```diff
@@ -559,3 +559,25 @@ def to_dict(self) -> Dict[str, Any]:  # skipcq: PYL-W0221
         else:
             json["trigger"] = self._trigger
         return json
+
+
+class SlackFile(JsonObject):
+    attributes = {"id", "url"}
+
+    def __init__(
+        self,
+        *,
+        id: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+        self._id = id
+        self._url = url
+
+    def to_dict(self) -> Dict[str, Any]:  # skipcq: PYL-W0221
+        self.validate_json()
+        json = {}
+        if self._id is not None:
+            json["id"] = self._id
+        if self._url is not None:
+            json["url"] = self._url
```

Comment:
If a user sets both, then both properties would be set. According to [the docs on this object](https://api.slack.com/reference/block-kit/composition-objects#slack_file), this would result in the payload being rejected:

> This [file](https://api.slack.com/types/file) must be an image and you must provide either the URL or ID. In addition, the user posting these blocks must have access to this file. If both are provided then the payload will be rejected.

Reply:
that's true but runtime error with server side error should be okay. i would like to avoid having duplicated logic on the client side for this (we already have a lot in this sdk but maintaining them is a concern I've been thinking about)